### PR TITLE
Create update yard bookings

### LINKED ISSUE #21: Yard Show Page - GET
When a host user in YS-FE visits a yard show page
A GET request is sent to YS-Engine
GET `/api/v1/yards/:id`
The json response must contain the following data:
```js
{
  "data": {
    "id": "1",
    "type": "yard",
    "attributes": {
"id": 14,  
"host_id": 1,  
"yard_name": “Hostname Backyard”,
"address": "123 Cool St",
"state": "CO",
"city": "Denver",
"zipcode": "80210",
“photo_url_1”: “https://howehowiehw”,
“photo_url_2”: “https://howehowiehw”,
“photo_url_3”: “https://howehowiehw”, 
"description": "Amazing yard with 300 sq ft of green grass and shady trees.",
"availability": "Only available on Monday, Wednesday, Friday from 3-7pm",
"payment": "Cold hard cash. No exceptions",
"price": "$25/hour",
"purpose_1": “pets”,
“purpose_2”: “events”
    }
  }
}
```

### LINKED ISSUE #42: Update Yard endpoint needs to be able to create and delete yard_bookings

### LINKED ISSUE #43: Create Yard Endpoint needs to create yard_bookings

### LINKED ISSUE #49: Add purposes to get yard endpoint

### PR DESCRIPTION
**Pull request by jbranham1**
Updated the Yard to create and delete yard purposes on create and update.
Added purposes to the yard serializer.
Created unique index on yard purposes table.

Close #43 
Close #42 
Close #49

### CODE REVIEW

File: app/controllers/api/v1/yards_controller.rb [hunk 1]
```diff
@@ -11,42 +11,65 @@ def show
...
+    else
+      error = "You must select at least one purpose"
+      render_error(error, :not_acceptable)
+    end
   end
 
   def destroy
     yard = Yard.find(params[:id])
     render json: Yard.destroy(params[:id])
   end
```
Comment (domo2192) [inline app/controllers/api/v1/yards_controller.rb:40]: Do we need something to clean yard purposes table as well when we destroy yards

**Review by domo2192 — APPROVED** [review]
It looks really good! You all have gotten really far into it already

File: app/controllers/api/v1/yards_controller.rb (hunk 1 above)
Comment (jbranham1) [reply app/controllers/api/v1/yards_controller.rb:40]: Talked about this in person!

**Review by abreaux26 — APPROVED** [review]

**Comment by abreaux26** [comment]
Nice Work!!

**Pull request merged**